pkg/R/SQLContext.R

```diff
@@ -1,5 +1,130 @@
 # SQLcontext.R: SQLContext-driven functions
 
+infer_type <- function(x) {
+  if (is.null(x)) {
+    stop("can not infer type from NULL")
+  }
+  type <- switch(class(x),
+                 integer = "long",
+                 character = "string",
+                 logical = "boolean",
+                 double = "double",
+                 numeric = "double",
+                 raw = "binary",
+                 list = "array",
+                 environment = "map",
+                 stop("Unsupported type for DataFrame"))
+
+  if (type == "map") {
```

Comment:
Can you add a short comment about the various null fields we have ? I see `valueContainsNull`, `containsNull`, `nullable` etc.